feat: company 생성 기능 구현

yejiscore opened the pull request:
## 📝 작업 내용
### 변경 사항
company 생성 기능 구현하였습니다.

<br>


### #️⃣ 연관된 이슈
closed #103

<br>

## PR 유형

이 PR은 어떤 변경 사항을 포함하고 있나요?

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [ ] UI 디자인 변경 (CSS 등)
- [ ] 코드에 영향을 주지 않는 변경사항 (오타 수정, 탭 사이즈 변경, 변수명 변경 등)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 또는 폴더명 수정
- [ ] 파일 또는 폴더 삭제
- [ ] 배포 준비, PR 관련 사항

<br>

## ✅ Check List

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [X] 변경 사항에 대한 테스트를 했습니다. (버그 수정/기능에 대한 테스트)
- [ ] CI/CD 파이프라인을 통과했습니다.

yejiscore committed:
refactor: company 생성 DTO 이름 변경

기능마다 dto 구분을 위해 기존의 CompanyRequestDto 를 CreateCompanyRequestDto 로 변경
(Response 도 동일)

Resolved #39

yejiscore committed:
feat: company 생성 기능 구현

company 생성 기능 구현하였습니다.

Resolved #103

nodajida commented:
넵 확인 했습니다!

Jungdaye89 commented on company/src/main/java/com/business/company/application/dto/response/CreateCompanyResponseDto.java:
```diff
@@ -0,0 +1,33 @@
+    private UUID hubId;
+    private UUID companyManagerId;
+
+    public static CreateCompanyResponseDto from(Company company) {
```
DTO 변환은 Mapper 클래스를 통해 통합 관리하기로 했습니다. 해당 변환 로직을 Mapper 클래스로 이동하는 방향으로 수정해주시면 좋을 것 같습니다!

AnJungGeon commented on company/src/main/java/com/business/company/domain/repository/CompanyRepository.java:
JPA 상속을 제거하고, 인프라 계층에서 관리하도록 해주세요.
도메인 계층에서는 JPA에 의존하지 않도록 Repository 인터페이스만 유지하는 것이 좋습니다 😄

yejiscore committed:
refactor: yml 파일 깃에 올라가지 않도록 수정

yejiscore committed:
refactor: mapper 추가 및 dto 수정

- mapper 를 추가하였습니다.
- 추가된 mapper와 맞게 dto랑 service도 수정하였습니다.

Resolved #103

yejiscore committed:
refactor: repository 계층 구조 변경

도메인 repository의 jpa 상속을 제거하고 infrastructure 를 생성하여 jpa 관리하도록 변경하였습니다.

Resolved #103

yejiscore committed:
fix: gitignore 수정

yml 파일 깃허브에 올라가지 않도록 수정하였습니다.

Resolved #103

yejiscore committed:
fix: use develop branch version of application.yml
